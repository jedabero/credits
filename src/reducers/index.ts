import { combineReducers } from '@reduxjs/toolkit';
import creditCards from './creditCards';

export default combineReducers({
  creditCards,
});
