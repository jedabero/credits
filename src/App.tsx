import { useCallback } from 'react';
import './App.css';
import { addCard, removeCard, selectAllCreditCards } from './reducers/creditCards';
import { useAppDispatch, useAppSelector } from './hooks';

function App() {
  const dispatch = useAppDispatch();
  const handleAdd = useCallback(() => dispatch(addCard({
    id: 1,
    name: 'test',
    quota: 24633,
    balance: 0,
    cutoffDay: 15,
    cardNumber: '***465',
  })), [dispatch]);
  const cards = useAppSelector(selectAllCreditCards);
  const handleDelete = useCallback(() => dispatch(removeCard(1)), [dispatch]);

  return (
    <div className="App">
      <main>
        <h6>Tasks</h6>
        <p>Definir Credito</p>
        <p>Definir Movimiento</p>
        <p>Ver valores detallados del mes y de cada movimiento</p>
        <p>{JSON.stringify(cards)}</p>
        <button type="button" onClick={handleAdd}>Add Credit Card</button>
        <button type="button" onClick={handleDelete}>Delete Credit Card</button>
      </main>
    </div>
  );
}

export default App;
