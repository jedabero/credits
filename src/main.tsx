import React from 'react';
import ReactDOM from 'react-dom/client';
import { Provider } from 'react-redux';
import { persistStore } from 'redux-persist';
import { PersistGate } from 'redux-persist/integration/react';
import App from './App';
import { store } from './store';
import './index.css';

const persistor = persistStore(store);

ReactDOM.createRoot(document.getElementById('root') as HTMLElement).render(
  <React.StrictMode>
    <Provider store={store}>
      <PersistGate loading={null} persistor={persistor}>
        <App />
      </PersistGate>
    </Provider>
  </React.StrictMode>,
);
