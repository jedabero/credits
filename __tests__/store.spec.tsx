jest.spyOn(console, 'error').mockImplementationOnce(() => {});

// eslint-disable-next-line import/first
import { store } from '../src/store';

describe('store test', () => {
  test('render App', () => {
    expect(store.getState()).toMatchSnapshot();
  });
});
