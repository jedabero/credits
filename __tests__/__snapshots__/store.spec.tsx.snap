// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`store test render App 1`] = `
{
  "creditCards": {
    "entities": {},
    "ids": [],
  },
  "movements": {
    "entities": {},
    "ids": [],
  },
}
`;
